src/main/java/org/folio/rest/impl/RequestsAPI.java

```diff
@@ -130,24 +129,14 @@ public void putRequestStorageRequestsByRequestId(
   }
 
   private Errors samePositionInQueueError(Request request) {
-    Error error = new Error();
-
-    error.withMessage("Cannot have more than one request with the same position in the queue")
-      .withAdditionalProperty("itemId", request.getItemId())
-      .withAdditionalProperty("position", request.getPosition());
-
-    List<Error> errorList = new ArrayList<>();
-    errorList.add(error);
-
-    Errors errors = new Errors();
-    errors.setErrors(errorList);
-
-    return errors;
+    return RequestsApiUtil
+      .samePositionInQueueError(request.getItemId(), request.getPosition());
   }
 
   private boolean isSamePositionInQueueError(AsyncResult<Response> reply) {
     return reply.succeeded()
```

Comment:
What happens if the action failed?

Reply:
I think the `isSamePositionInQueueError` method will return `false` in this case.

I did not modify original logic of this method, I just replaced `reply.result().getEntity().toString().contains("request_itemid_position_idx_unique");` with `RequestsApiUtil.hasSamePositionConstraintViolated(reply.result().getEntity().toString())`, so I can reuse it in RequestsBatchApi.

Reply:
Ok, if a failure will trigger a 500 error message, then I'm ok with that. Thank you for sharing an explanation with me.